tests/python_package_test/test_engine.py

```diff
@@ -1505,42 +1504,129 @@ def test_get_split_value_histogram(self):
+                                              first_metric_only, eval_train_metric):
+            params = {
+                'objective': 'regression',
+                'learning_rate': 0.9,
+                'num_leaves': 10,
+                'metric': metric_list,
+                'verbose': -1,
+                'seed': 123,
+                'gpu_use_dp': True
+            }
+            ret = lgb.cv(dict(params, first_metric_only=first_metric_only),
+                         stratified=False,
+                         train_set=lgb_train,
+                         num_boost_round=25,
+                         early_stopping_rounds=5, verbose_eval=verbose_eval,
+                         eval_train_metric=eval_train_metric)
+            self.assertEqual(assumed_iteration, len(ret[list(ret.keys())[0]]))
+
+        best_iter_l1 = 16
+        best_iter_l2 = 19
+        best_iter_min = min([best_iter_l1, best_iter_l2])
+        metrics_combination_train_regression([], best_iter_l2, False)
+        metrics_combination_train_regression([], best_iter_l2, True)
+        metrics_combination_train_regression(None, best_iter_l2, False)
+        metrics_combination_train_regression(None, best_iter_l2, True)
+        metrics_combination_train_regression('l2', best_iter_l2, True)
+        metrics_combination_train_regression('l1', best_iter_l1, True)
+        metrics_combination_train_regression(['l2', 'l1'], best_iter_l2, True)
+        metrics_combination_train_regression(['l1', 'l2'], best_iter_l1, True)
+        metrics_combination_train_regression(['l2', 'l1'], best_iter_min, False)
+        metrics_combination_train_regression(['l1', 'l2'], best_iter_min, False)
+
+        best_iter_l1 = 6
+        best_iter_l2 = 11
+        best_iter_min = min([best_iter_l1, best_iter_l2])
+        metrics_combination_cv_regression(None, best_iter_l2, True, False)
+        metrics_combination_cv_regression('l2', best_iter_l2, True, False)
+        metrics_combination_cv_regression('l1', best_iter_l1, True, False)
+        metrics_combination_cv_regression(['l2', 'l1'], best_iter_l2, True, False)
+        metrics_combination_cv_regression(['l1', 'l2'], best_iter_l1, True, False)
+        metrics_combination_cv_regression(['l2', 'l1'], best_iter_min, False, False)
+        metrics_combination_cv_regression(['l1', 'l2'], best_iter_min, False, False)
+        metrics_combination_cv_regression(None, best_iter_l2, True, True)
+        metrics_combination_cv_regression('l2', best_iter_l2, True, True)
+        metrics_combination_cv_regression('l1', best_iter_l1, True, True)
+        metrics_combination_cv_regression(['l2', 'l1'], best_iter_l2, True, True)
+        metrics_combination_cv_regression(['l1', 'l2'], best_iter_l1, True, True)
+        metrics_combination_cv_regression(['l2', 'l1'], best_iter_min, False, True)
+        metrics_combination_cv_regression(['l1', 'l2'], best_iter_min, False, True)
+
+        # composite name metrics(lambdarank)
```

Comment:
Seems that this test case is not needed at all, since we do not perform any operations or checks for metric name now. We don't care about its' name at Python side.